community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/index/storage/FailureStorage.java

```diff
@@ -115,6 +115,10 @@ public synchronized void storeIndexFailure( String failure ) throws IOException
         File failureFile = failureFile();
         try ( StoreChannel channel = fs.open( failureFile, "rw" ) )
         {
+            byte[] existingData = new byte[(int) channel.size()];
+            channel.read( ByteBuffer.wrap( existingData ) );
+            channel.position( lengthOf( existingData ) );
```

Comment:
We can still run into `no space left on device` failures with a file that's preallocated with zeros. File systems that support sparse files will likely not actually write the zeros down, but just note how many there are. Similar problem with file systems that use compression. Even worse with file systems that do copy-on-write, since then even overwriting a file filled with random data can cause out of space failures. But I guess it's better than nothing; I don't think we can reasonably guard for all possible scenarios that can lead to running out of space.